File: server/api/taskgroup.go
package api

type TaskGroup struct {
	Id        uint    `json:"id" form:"id"`
	ProjectId uint    `json:"projectId" form:"projectId"`
	Name      string  `json:"name" form:"name"`
	Serial    int     `json:"serial" form:"serial"`
	Tasks     []*Task `json:"tasks" form:"tasks"`
}

type MoveTaskGroupResp struct {
	Serial     int   `json:"serial" form:"serial"`
	UpdateAll  bool  `json:"updateAll" form:"updateAll"`
	AllSerials []int `json:"allSerials" form:"allSerials"`
}

type MoveTaskGroupReq struct {
	ProjectId uint `json:"projectId" form:"projectId"`
	Id        uint `json:"id" form:"id"`     // 要移动的id
	Prev      uint `json:"prev" form:"prev"` // 移动之后前面的id
	Next      uint `json:"next" form:"next"` // 移动之后后面的id
}

File: web/src/api/taskgroup.ts
import http from './axios'
import { Task } from './task'

export interface TaskGroup {
  id?: number
  projectId?: number
  name?: string
  serial?: number
  tasks?: Task[]
  color?: string
}

const url = '/api/taskgroup'
export const addTaskGroup = (t: TaskGroup) => {
  return http({
    method: 'POST',
    url: url,
    data: t,
  }).then((res) => res.data)
}

export const updateTaskGroup = (t: TaskGroup) => {
  return http({
    method: 'PUT',
    url: url,
    data: t,
  }).then((res) => res.data)
}

export const delTaskGroup = (t: TaskGroup) => {
  return http({
    method: 'DELETE',
    url: url,
    data: { id: t.id },
  }).then((res) => res.data)
}

export const moveTaskGroup = (req: {
  id: number
  projectId: number
  prev?: number
  next?: number
}) => {
  return http({
    method: 'POST',
    url: url + '/move',
    data: req,
  }).then((res) => res.data)
}
